lib/system/io.nim

```diff
@@ -318,6 +328,40 @@ proc getFileHandle*(f: File): FileHandle =
   ## Use `getOsFileHandle` instead.
   c_fileno(f)
 
+proc duplicateFileHandle*(oldfd: FileHandle): FileHandle =
+  ## Return a copy of the file handle oldfd.
+  ## After a successful return, both FileHandle may be used interchangeably.
+  ## They refer to the same open file description and share file offset and status flags.
+  ## Calls POSIX function `dup` on Linux and  `_dup` on Windows
+  let retValue = c_dup(oldfd)
+  if retValue != -1:
+    retValue
+  else:
+    when not defined(NimScript):
+      # Raise
+      var e: ref Exception
+      new(e)
+      e.msg = $strerror(errno)
+      raise e
+    else:
+      # Same as checkErr
+      quit(1)
+
+proc assignFileHandle*(oldfd: FileHandle, newfd: FileHandle) =
```

Comment:
I'd call this: `duplicateTo`. We don't need the `FileHandle` within the name since `FileHandle` is a distinct type already.

Reply:
Wouldn't `duplicate` create confusion since `dup` already exists in the language ?

Reply:
`duplicateTo` is different enough from `dup`, nim doesn't define gratuitous aliases